eq-author-api/src/validation/customKeywords/validateRoutingLogicalAND.js

```diff
@@ -0,0 +1,137 @@
+const { groupBy } = require("lodash");
+const { getAnswerById } = require("../../../schema/resolvers/utils");
+const {
+  CURRENCY,
+  NUMBER,
+  PERCENTAGE,
+  UNIT,
+} = require("../../../constants/answerTypes");
+const createValidationError = require("../createValidationError");
+const { ERR_LOGICAL_AND } = require("../../../constants/validationErrorCodes");
+
+const getRHS = e =>
+  e.right && e.right.customValue && e.right.customValue.number;
+
+const areUnanswered = e => e.condition === "Unanswered";
+
+module.exports = ajv => {
+  ajv.addKeyword("validateRoutingLogicalAND", {
+    validate: function isValid(
+      otherFields,
+      { expressions },
+      fieldValue,
+      dataPath,
+      parentData,
+      fieldName,
+      questionnaire
+    ) {
+      const invalidAnswerIds = new Set();
+      const expressionsByAnswerId = groupBy(expressions, "left.answerId");
+      const potentialConflicts = Object.entries(expressionsByAnswerId).filter(
+        ([answerId, expressions]) =>
+          expressions.length > 1 && answerId && answerId.length
+      );
+
+      const error = answerId => invalidAnswerIds.add(answerId);
+
+      potentialConflicts.forEach(([answerId, expressions]) => {
+        // Handle invalid combination of "Unanswered" with any answer-requiring condition
+        if (
+          expressions.some(areUnanswered) &&
+          !expressions.every(areUnanswered)
+        ) {
+          return error(answerId);
+        }
+
+        // Bail out if answer isn't numerical - remaining code validates number-type answers
+        const answer = getAnswerById({ questionnaire }, answerId);
+        if (![CURRENCY, NUMBER, UNIT, PERCENTAGE].includes(answer.type)) {
+          return;
+        }
+
+        // Calculate precision of numerical answer based on trailing decimal count
+        const precision = Math.pow(10, -answer.properties.decimals);
+
+        const equalitySet = new Set();
+        const nonequalitySet = new Set();
+        let lowerLimit = -Infinity;
+        let upperLimit = Infinity;
+
+        for (const expression of expressions) {
+          let rhs = getRHS(expression);
```

Comment:
What does`rhs` stand for? My only comment here is that it could be renamed to be more obvious.

Reply:
Right hand side :-) So if you have a "x > 40" rule for instance 40 is the right hand side value.

I'll change it tomorrow and stick some extra comments in.

Reply:
I've re-named it to `rightHandSide` - actually didn't need the `getRHS` function either so that's been inlined now instead - and added/changed a few comments.

Let me know if you think any part needs more elaboration.

Reply:
Ahh amazing. Thanks for clarifying and elaborating on that in the code.